packages/next/src/server/app-render/create-error-handler.tsx

```diff
@@ -173,6 +172,9 @@ export function createHTMLErrorHandler(
     // If this is a navigation error, we don't need to log the error.
     if (isNextRouterError(err)) return err.digest
 
+    // In SSR rendering, we always collect the error
+    reactServerErrors.set(err.digest, err)
```

Comment:
This is very confusing b/c we're stashing errors in this map that aren't necessarilyl React server errors. We should just track them separately